File: APIs/AuthService/AuthService/DTOs/AttendedCategory.cs
using Newtonsoft.Json;

namespace AuthService.DTOs
{
    public class AttendedCategory
    {
        [JsonProperty("sportCategory")]
        public SportCategoryObject SportCategory { get; set; } = null!;
        [JsonProperty("honors")]
        public int Honors { get; set; } 
    }
}

File: SocialAppFlutter/social_app/lib/Models/attended_category.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:social_app/Models/sport_category.dart';

part 'json_generated_code/attended_category.g.dart';

@JsonSerializable(explicitToJson: true)
class AttendedCategory {
  final SportCategory sportCategory;
  int honors;

  AttendedCategory(this.sportCategory, this.honors);

  factory AttendedCategory.fromJson(Map<String, dynamic> json) =>
      _$AttendedCategoryFromJson(json);

  Map<String, dynamic> toJson() => _$AttendedCategoryToJson(this);
}
